vm/chef/cookbooks/singlefs/files/singlefs-validate

```diff
@@ -0,0 +1,250 @@
+readonly ENABLE_NFS="$(get_metadata instance/attributes/ENABLE_NFS)"
+readonly ENABLE_SMB="$(get_metadata instance/attributes/ENABLE_SMB)"
+readonly FILE_SYSTEM="$(get_metadata instance/attributes/FILE_SYSTEM)"
+readonly STORAGE_POOL="$(get_metadata instance/attributes/STORAGE_POOL_NAME)"
+readonly DISABLE_MONITORING="$(get_metadata instance/attributes/DISABLE_MONITORING || true)"
+readonly VIRTUAL_IP="$(get_metadata instance/attributes/VIRTUAL_IP || true)"
+for variable in STORAGE_POOL ENABLE_NFS ENABLE_SMB FILE_SYSTEM; do
+  eval "value=$(echo \${$variable})"
+  if [[ -z "$value" ]]; then
+    echo "Required metadata variable '$variable' not set. Aborting validation."
+    exit 1
+  fi
+done
+
+echo "File system:"
+case "${FILE_SYSTEM}" in
+  zfs)
+    echo -n "Test that the ZFS kernel module is installed ... "
+    validate "lsmod | grep zfs > /dev/null"
+    ;;
+  btrfs)
+    echo -n "Test that the BtrFS kernel module is installed ... "
+    validate "lsmod | grep btrfs > /dev/null"
+    ;;
+  xfs)
+    echo -n "Test that the XFS kernel module is installed ... "
+    validate "lsmod | grep xfs > /dev/null"
+    ;;
+  ext4)
+    echo "No tests for ext4."
+    ;;
+esac
+
+echo -n "Test that a ${FILE_SYSTEM} file system was created ... "
+case "${FILE_SYSTEM}" in
+  zfs)
+    validate "zpool list | grep -v 'no pools available' > /dev/null"
+    ;;
+  btrfs)
+    validate "btrfs filesystem show '${STORAGE_POOL}' > /dev/null"
+    ;;
+  xfs)
+    validate "findmnt -n -t xfs '/${STORAGE_POOL}' > /dev/null"
+    ;;
+  ext4)
+    validate "findmnt -n -t ext4 '/${STORAGE_POOL}' > /dev/null"
+    ;;
+esac
+
+
+# Test file creation/deletion on primary only.
+if is_primary; then
+  test_file "/${STORAGE_POOL}/basic_test.txt"
+fi
+
+# (Optional) NFS testing.
+if [[ "${ENABLE_NFS}" == "enable:True" ]]; then
+  echo -e "\nNFS:"
+  test_nfs "${HOST_IP}"
+else
+  echo -e "\nNFS disabled. Skipping tests."
+fi
+
+# (Optional) Samba testing.
+if [[ "${ENABLE_SMB}" == "enable:True" ]]; then
+  echo -e "\nSamba:"
+  echo -n "Test that Samba server is running ... "
+  validate "ps aux | grep -v grep | grep smbd  > /dev/null"
+
+  echo -n "Test that Samba can be mounted ... "
+  validate "mount -t cifs '//${HOST_IP}/${STORAGE_POOL}' /mnt -o username=guest"
+
+  # Test file creation/deletion on primary only.
+  if is_primary; then
+    test_file /mnt/smb_test.txt
+  fi
+
+  echo -n "Test that Samba can be unmounted ... "
+  validate "umount /mnt"
+else
+  echo -e "\nSMB disabled. Skipping tests."
+fi
+
+# (Optional) Virtual IP NFS testing.
+if [[ -n "${VIRTUAL_IP}" ]]; then
+  if [[ "${ENABLE_NFS}" == "enable:True" ]]; then
+    echo -e "\nNFS over virtual IP:"
+    test_nfs "${VIRTUAL_IP}"
+  fi
+else
+  echo -e "\nVirtual IP not configured. Skipping tests."
+fi
+
+if is_debian; then
+  APACHE2_SERVICE="apache2"
+fi
+if is_centos; then
+  APACHE2_SERVICE="httpd"
+fi
+
+
+# Test monitoring stack on primary
+if [[ -n "${DISABLE_MONITORING}" ]]; then
```

Comment:
**(Postpone action):** `DISABLE_MONITORING` attribute does not exist. Can be removed?